### PR TITLE
Improves SearchResult component

### PR DESCRIPTION
**Pull request by schalkventer**
image

image

### CODE REVIEW

File: _includes/components/search-results/SearchResult/partials/LandingLayout.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        <div className="SearchResult-leftHeading">
+          <div className="Page-title u-margin0">Found in {year} {tabOptions[tab]}</div>
+        </div>
+        {count > 3 ? viewAll(updateTabWrap, count) : null}
+      </div>
+    </div>
+  );
+};
+
+const fallback = {
```
Comment (jbothma) [inline _includes/components/search-results/SearchResult/partials/LandingLayout.jsx]: is this getting used or just an accidental leftover?

**Comment by jbothma** [comment]
when I click on a year to show more provincial results I get more national results

**Comment by jbothma** [comment]
when I select a year under national or provincial to show more, the sphere tab is not indicated as the selected tab

**Comment by jbothma** [comment]
"Source" links to PDFs open in the current tab and not a new one. I think they should open in a new tab as elsewhere

**Comment by jbothma** [comment]
when I open source PDF links in the chrome mobile emulator they don't open - not sure if that's the emulator. not sure if it'd be fixed by making them open in a new tab. let's wait and see.

File: _includes/components/search-results/SearchResult/partials/buildNotice.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+const calcNoticeObj = (amount, error) => {
+  if (error) {
+    return {
+      size: 'is-1of1',
+      title: 'Something went wrong.',
+      text: 'Please try again at a later point.',
+    };
+  }
+
+  switch (amount) {
```
Comment (jbothma) [inline _includes/components/search-results/SearchResult/partials/buildNotice.jsx]: I like this but I don't see it working
vulekamali-improved-search-missing-help-result

**Comment by jbothma** [comment]
When I use the classic financial year selector to change year on the search landing page, the page reloads using the previously-selected financial year, not the newly-selected financial year.

**Comment by jbothma** [comment]
When I select another financial year using the buttons showing number of results per financial year per sphere on landing page, the URL changes but the results shown are for the previously-selected financial year.

This is why I strongly believe that a single page app which updates the URL should derive its state from the URL. If you derive state from the URL, it automatically does the right thing regardless of whether you navigated there by clicks or whether you used a stored URL. It also avoids this bug.

**Comment by jbothma** [comment]
When I click on See All n Results above the provincial section, the view changes and my viewport stays at the bottom of the page. That's confusing because I might not know exactly where I am, and I probably want to start at the top where the most relevant results are.

**Comment by jbothma** [comment]
when I'm on a specific sphere search results tab and change financial year using the selector at the top, I get the same results back (for the previously-selected financial year)

File: _includes/components/search-results/SearchResult/partials/getFacetResults.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import fetchWrapper from './../../../../utilities/js/helpers/fetchWrapper.js';
+import normaliseReturn from './normaliseReturn.js';
+import highlightResults from './highlightResults.js';
+import createPromiseToken from './../../../../utilities/js/helpers/createPromiseToken.js';
+
+
+export default function getFacetResults(phrase, facet, start = 0) {
+  const request = new Promise((resolve, reject) => {
+    const innerRequest = fetchWrapper(`https://data.vulekamali.gov.za/api/3/action/package_search?q=${encodeURI(phrase)}&start=${start}&rows=5&fq=+organization:national-treasury+vocab_financial_years:2018-19+extras_department_name_slug:[*%20TO%20*]+extras_geographic_region_slug:[*%20TO%20*]+vocab_spheres:${facet}&ext_highlight=true`);
```
Comment (jbothma) [inline _includes/components/search-results/SearchResult/partials/getFacetResults.js]: hardcoded year :)

File: _includes/components/search-results/SearchResult/partials/getLandingResults.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        count,
+        name,
+        url,
+      };
+    });
+  };
+
+  const request = new Promise((resolve, reject) => {
+    const urlsArray = [
+      `https://data.vulekamali.gov.za/api/3/action/package_search?q=${encodeURI(phrase)}&start=0&rows=3&fq=+organization:national-treasury+vocab_financial_years:2018-19+extras_department_name_slug:[*%20TO%20*]+extras_geographic_region_slug:[*%20TO%20*]+vocab_spheres:national&ext_highlight=true`,
```
Comment (jbothma) [inline _includes/components/search-results/SearchResult/partials/getLandingResults.js]: hardcoded years

File: _includes/components/search-results/SearchResult/partials/getFacetResults.js (hunk 3 above)
Comment (schalkventer) [reply _includes/components/search-results/SearchResult/partials/getFacetResults.js]: Good catch. 👍

**Comment by jbothma** [comment]
On mobile I have no way of seeing all results since the "Show more result" year buttons take me to the landing page.

**Comment by jbothma** [comment]
show more doesn't work http://localhost:4000/2015-16/search-result?search=expropriation&view=provincial

**Comment by jbothma** [comment]
when I'm on a province detail page, it's surprising that changing year using the year selector takes me back to the landing page. I'd be satisfied with just adding something to the icebox to investigate a more comfortable behaviour

File: _includes/components/search-results/SearchResult/partials/createLinkText.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export default function (innerTabKey) {
+  switch (innerTabKey) {
```
Comment (jbothma) [inline _includes/components/search-results/SearchResult/partials/createLinkText.js]: I'd recommend using a variable called sphere at this point - something more semantic and less tied to the UI. That'd force that variable to be scoped appropriately and avoid weird things happening when we add a tab that isn't a sphere. Just a tip, not a change request for this PR

File: _includes/components/search-results/SearchResult/partials/normaliseReturn.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import normaliseItem from './normaliseItem.js';
+
+
+export default function normaliseReturn(returnObj) {
```
Comment (jbothma) [inline _includes/components/search-results/SearchResult/partials/normaliseReturn.js]: I have absolutely no idea what this module does. Does it always normalise the response from package_search to the same internal form? If so, we could really call the module NormalisePackageSearchReturn or have a comment hint to the service it's about.

File: _includes/components/search-results/SearchResult/partials/createLinkText.js (hunk 5 above)
Comment (schalkventer) [reply _includes/components/search-results/SearchResult/partials/createLinkText.js]: Good suggestion. Will change in CSO dataset branch.

File: _includes/components/search-results/SearchResult/partials/normaliseReturn.js (hunk 6 above)
Comment (schalkventer) [reply _includes/components/search-results/SearchResult/partials/normaliseReturn.js]: This and `normaliseItem` can actually be merged into one module. In short, that would then just be a module that takes the AJAX response and turns it into a object that's more semantic to the front-end/UI. For example:

image

Comment (schalkventer) [reply _includes/components/search-results/SearchResult/partials/normaliseReturn.js]: Renamed the module to `normaliseServerResponse`

**Pull request merged**